lab/homework/index.js

```diff
@@ -0,0 +1,148 @@
+export class HomeworkError extends Error {
+    
+    static ERROR_CODE = {
+        NOT_A_PRODUCT: 'NOT A PRODUCT',
+        NOT_IN_CART: 'NOT IN CART',
+        EXPIRED: 'EXPIRED',
+        POOR_ALERT: 'NO MONEY'
+    };
+
+    static ERROR_MESSAGE = {
+        [HomeworkError.ERROR_CODE.NOT_A_PRODUCT]: 'The method must receive an instance of product!',
+        [HomeworkError.ERROR_CODE.NOT_IN_CART]: 'The product must exist in the cart!',
+        [HomeworkError.ERROR_CODE.EXPIRED]: 'The aliment is expired!',
+        [HomeworkError.ERROR_CODE.POOR_ALERT]: 'The customer does not have enough money!',
+    };
+    
+    constructor({ code, message = HomeworkError.ERROR_MESSAGE[code] }) {
+        super(message);
+        this.code = code;
+    }
+    
+    toString() {
+        return this.message + '\n' + this.stack;
+    }
+    
+}
+
+
+
+export class Product {
+    
+    constructor({name, price}) {
+        this.name = name;
+        this.price = price;
+    }
+    
+}
+
+
+
+export class Aliment extends Product {
+
+    constructor({name, price, expirationDate, grammage}) {
+        super({name, price});
+        this.expirationDate = expirationDate;
+        this.grammage = grammage;
+    }
+    
+    isExpired() {
+        // compare expiration date with current date
+        return this.expirationDate < new Date();
+    }
+
+}
+
+
+
+export class Furniture extends Product {
+
+    constructor({name, price, colour}) {
+        super({name, price});
+        this.colour = colour;
+    }
+
+}
+
+
+
+export class Person {
+    
+}
+
+
+
+export class Customer extends Person {
+    
+    constructor({money, shoppingCart}) {
+        super({});
+        this.money = money;
+        this.shoppingCart = shoppingCart;
+    }
+    
+    addToCart(product) {
+        if (product instanceof Product) {
+            if(product instanceof Aliment && product.isExpired()) {
+                throw new HomeworkError({
+                    code: HomeworkError.ERROR_CODE.EXPIRED
+                });
+            }
+            // add the product to the cart
+            this.shoppingCart.push(product);
+        } else {
+            throw new HomeworkError({
+                code: HomeworkError.ERROR_CODE.NOT_A_PRODUCT
+            });
+        }
+    }
+    
+    removeFromCart(product) {
+        if (product instanceof Product) {
+            // find element in shopping cart
+            var idx = this.shoppingCart.findIndex((el) => el.toString() === product.toString())
```

Comment:
There are two problems here:

1. Never use `var` in your code. Use only `let` or `const`. 
In this case `const` is more appropriate since you don't plan to change its value.

2. When you call `.toString()` method on an instance of a class that doesn't have this method implemented it will use the one from the `Object` (implicit parent) class which returns the primitive string `[object Object]`.
In your case all your products will appear to be equal, so you will always remove the first one from that Array. 
For simplicity, you could directly use `===` to compare object references.